### PR TITLE
Add rule S6859 (`eslint-plugin-import/no-absolute-path`): Imports should not use absolute paths

### LINKED ISSUE #4432: Create rule S6859: Imports should not use absolute paths
[eslint-plugin-import/no-absolute-path](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-absolute-path.md)

[RSPEC PR](https://github.com/SonarSource/rspec/pull/3465)

[implementation PR](https://github.com/SonarSource/SonarJS/pull/4439)

### PR DESCRIPTION
**Pull request by ilia-kebets-sonarsource**
Fixes #4432

### CODE REVIEW

**Comment by ilia-kebets-sonarsource** [comment]
@yassin-kammoun-sonarsource Could you please run `rule-api`?
We can merge this without squashing so we can highlight how we add a new plugin.

File: package.json [hunk 1]
```diff
@@ -81,6 +81,7 @@
     "builtin-modules": "3.3.0",
     "bytes": "3.1.2",
     "eslint": "8.47.0",
+    "eslint-plugin-import": "^2.29.0",
```
Comment (yassin-kammoun-sonarsource) [inline package.json:84]:
```suggestion
    "eslint-plugin-import": "2.29.0",
```

Comment (yassin-kammoun-sonarsource) [reply package.json:84]: Don't forget to update the `package-lock.json` file.

**Review by yassin-kammoun-sonarsource — APPROVED** [review]

File: packages/jsts/src/linter/quickfixes/messages.ts [hunk 2]
```diff
@@ -28,6 +28,7 @@ const quickFixMessages = new Map<string, string>([
   ['comma-dangle', 'Remove this trailing comma'],
   ['eol-last', 'Add a new line at the end of file'],
   ['jsx-no-useless-fragment', 'Remove redundant fragment'],
+  ['no-absolute-path', 'Do not import modules using an absolute path'],
```
Comment (yassin-kammoun-sonarsource) [inline packages/jsts/src/linter/quickfixes/messages.ts:31]:
```suggestion
  ['no-absolute-path', 'Replace with a relative path'],
```

**Review by yassin-kammoun-sonarsource — CHANGES_REQUESTED** [review]

**Review by yassin-kammoun-sonarsource — APPROVED** [review]

**Pull request merged**